File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        int i=0;
         int c=0;
        int mcnt=0;
        for(int j=0;j<s.length();j++){
            char ch=s.charAt(j);
            if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u') c++;
            if(j-i+1==k){
                mcnt=Math.max(mcnt,c);
                char cha=s.charAt(i);
                 if(cha=='a' || cha=='e' || cha=='i' || cha=='o' || cha=='u') c--;
                i++;
            }
        }
    return mcnt;
    }
}   

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        count =0
        mcount=0
        for i in range(0,k):
            if s[i] in 'aeiou':
                count+=1
        mcount=count
        for i in range(k,len(s)):
            if(s[i-k] in 'aeiou'):
                count-=1
            if(s[i] in 'aeiou'):
                count+=1
            if(count>mcount):
                mcount=count
        return mcount
            
        